commons/c2cgeoportal_commons/alembic/static/53d671b17b20_add_timezone_on_datetime_fields.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2017-2018, Camptocamp SA
+# All rights reserved.
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# The views and conclusions contained in the software and documentation are those
+# of the authors and should not be interpreted as representing official policies,
+# either expressed or implied, of the FreeBSD Project.
+
+"""Add timezone on datetime fields
+
+Revision ID: 53d671b17b20
+Revises: 1857owc78a07
+Create Date: 2018-05-29 08:44:17.675988
+"""
+
+from alembic import op
+from c2cgeoportal_commons.config import config
+
+# revision identifiers, used by Alembic.
+revision = '53d671b17b20'
+down_revision = '1857owc78a07'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    staticschema = config['schema_static']
+
+    op.execute('''
+SET TIME ZONE 'UTC';
+ALTER TABLE {staticschema}.user ALTER COLUMN last_login TYPE timestamp with time zone;
```

Comment:
I test it but I get the wrong time (shifted of 2 hours), with old and new logins ...

Reply:
Yes, it is the purpose of this pull request.

Reply:
Note the +2 at the end of the ISO format in admin UI.
Before datetimes were stored and displayed in UTC.
Now timezones are stored in UTC and displayed in UTC+2.

Reply:
In the admin interface I get `2018-05-30 12:57:22.058844` (table) or `2018-05-30 12:57:22` (form) But I login at `2018-05-30 14:57:22 GMT+2` ...

Reply:
With my PR you should get 2018-05-30 14:57:22 GMT+2 in admin UI.
Note that pg_admin will show you the date in local time as postgresql already return it in local time.

Reply:
I do a mistake in my test, and it's effectively working :-)

Reply:
By creating fields as timestamp with timezone, and creating python datetime objects with a timezone, as datetimes are timezone aware we do not have to worried about timezone afterward and everything should work as expected.

Reply:
```
>>> d_utc = datetime.now(pytz.utc)
>>> d_local = d_utc.astimezone(pytz.timezone('Europe/Paris'))
>>> print(d_utc - d_local)
0:00:00
```
When datetimes are time zone aware, all should works as expected !
No ambiguity in comparisons, no ambiguity on what we have in database, no ambiguity in database request and results.
It like coordinates with a projection system.

Reply:
but: 
```
>>> d1 = datetime.now()
>>> d2 = datetime.utcnow()
>>> print(d1-d2)
2:00:00
```
This mean that it change the comportment...